sdk/storage/azure-storage-blob/CHANGELOG.md

```diff
@@ -2,6 +2,10 @@
 
 ## 12.9.0b1 (Unreleased)
 
+## 12.8.2 (2021-06-25)
+**Fixes**
+- Fixed the auto decompress response body bug, and let user opt in decompress or not.
```

Comment:
is it a bug or new functionality?
it seems that decompress is true by default - should this be "Added ability to opt-out from decompression" ?

Reply:
great question! I feel it's more like a bug though there's api change... because currently if the blob/file content is gzipped and content size is larger than 32MB, user are never be able to download the content back..... changing the api is a workaround to fix this bug, the ideal behavior is to download whatever uploaded (decompress is ideally set to False), while switching decompress to True will cause a breaking change, because for those blobs whose size is less or equal to 32MB are able to download the blob unzipped.
Let me know if we want to make it a feature.

Reply:
regarding this `it seems that decompress is true by default - should this be "Added ability to opt-out from decompression" ?`
sounds great, I will rephrase it in the way you recommended

Reply:
Given that extra context. I agree this is a bug.
I also suggest we internally force underlying http client to not decompress without adding this option to public surface.

Reply:
Then It think it will be a breaking change for the users whose blob is less than 32MB and relying on the auto decompress feature.

Reply:
That'd be users who accidentally discovered this behavior, i.e. users who uploaded already compressed bytes and set content-encoding header and then expecting uncompressed payload to get back from storage. This is quite a corner case - do we believe such user exist? I'd rather expect them to fill a bug rather than leverage this - they must be surprised as well.

Reply:
@annatisch what do you think ?

Reply:
Imo we should fix this trap and don't decompress by default. If by any chance users leveraging that exists we may consider having a flag for them and documenting the behavior change.

Reply:
Thanks @kasobol-msft 
We should keep the current behaviour and auto-decompress by default is the content-encoding header is present.
This is the standard behaviour across Python HTTP libraries, and would be the expectation of a typical Python customer.
Granted, this default is different to the other languages - but is idiomatic to Python.

Reply:
@annatisch I'm not sure if this is expected/idiomatic. When I look at https://github.com/Azure/azure-storage-python/issues/509 and referenced issues it seems that users are surprised that high level API is attempting to decompress blob and are putting workaround in place.
They don't interact with low level HTTP libraries but our client API. Why low level component we use internally should dictate how our download api behaves? 
Besides it's kind of bad experience if roundtrip (client.upload -> client.download) just doesn't work if someone is setting content-encoding on blob.

Reply:
@annatisch I agree with Kamil here, this will make a fundamental difference in behavior between out Python and other libraries. I am ok with an option to download it and decompress it, we should not be trying to interpret what the user's intention is. The user might want to validate integrity before decompress (make sure someone has not replaced the binary by checking against a stored SHA). Lets sync up offline and we can post the resolution here.